La formazione per autodidatti

```diff
@@ -0,0 +1,77 @@
+Inutile dire che di sola romanticheria non si vive e che spesso e volentieri, anche utilizzando gli strumenti più appropriati e più adatti per sollevarci da lavori ripetitivi, la vita quotidiana di un programmatore non è sistematicamente soggetta a scariche di adrenalina.
+Anzi, dietro un lavoro ben fatto, di norma, c’è tanta, tanta “falegnameria”.
+Insomma, se alle domande “ho una reale predisposizione?” e “sono disposto ad accettare che il percorso di istruzione che sto per intraprendere non sarà tutto un tappeto di petali di rose?” avete risposto con un SI netto e sicuro, siete già a metà dell’opera, congratulazioni!
+
+Tutti possiamo programmare, allora?
+In teoria, nel momento stesso in cui pensate “ah, ora vado a mettere su la caffettiera”, voi tutti avete in mente –senza rendervene conto-, un piano precisissimo che consiste nei passi di 
+
+1)	Prendere la caffettiera
+2)	Riempire d’acqua la caldaia fino al segno
+3)	Inserire il filtro
+4)	Svuotare dell’eccesso di acqua
+5)	Riempire il filtro con il caffè senza pressarlo
+6)	Ecc…. fino alla tazzina fumante
+
+Non tutti i problemi si risolvono altrettanto facilmente ma spesso, più che la competenza tecnica, quello che fa un’enorme differenza è l’approccio al problema e –di riflesso-, anche alla sua soluzione.
+Esempio banale ma calzante: una volta mi è capitato di avere a che fare con un’applicazione discretamente complessa nella quale, ad un certo punto del processo, avrei dovuto stabilire in un loop molto pesante se nel database dovevo effettuare un’operazione di INSERT o di UPDATE.
+Richiedendo la cosa molti controlli e volendola semplificare a tutti i costi, risolsi felicemente la faccenda inizializzando valori nelle tabelle all’apertura dell’applicazione.
+E ciò, fece in modo che avessi automaticamente ed a gratis, solo operazioni di UPDATE e quindi potei eliminare tutta una serie di controlli lunghi, complessi, farraginosi e forieri di errori.
+Non è un esempio esaustivo, ma spero vi abbia suggerito un’idea migliore di quella che poche righe fa ho chiamato “creatività”.
+
+Tra l’altro, l’esempio che vi ho appena fornito, mette in luce che la programmazione, sia pure con una buona dose di creatività, richiede comunque ed in prima istanza un **approccio analitico ai problemi** e a seguire, la conoscenza dei metodi/mezzi/strumenti con cui implementare una possibile soluzione (perché si: “c’è sempre più di un modo per farlo”).
+Fortunatamente, è possibile migliorare le proprie capacità analitiche attraverso un lungo periodo di esercitazione mentre per la creatività… o ce l’hai o non ce l’hai, punto.
```

Comment:
Viene spontaneo definire alcune capacità come "innate". E se, come Carol Dweck ci insegna, alcune capacità fossero più presenti semplicemente perché la persona ha fatto più pratica con quella skill? 
In questo modo non trasmetteremo che una persona non creativa non possa diventarlo. Sempre che ovviamente sia una sua priorità migliorare questo aspetto ed è disposta a fare pratica.